pynear/include/crc32.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstddef>
+#include <stdint.h>
+
+struct crc32 {
+    static void generate_table(uint32_t (&table)[256]) {
+        uint32_t polynomial = 0xEDB88320;
+        for (uint32_t i = 0; i < 256; i++) {
+            uint32_t c = i;
+            for (size_t j = 0; j < 8; j++) {
+                if (c & 1) {
+                    c = polynomial ^ (c >> 1);
+                } else {
+                    c >>= 1;
+                }
+            }
+            table[i] = c;
+        }
+    }
+
+    static uint32_t update(uint32_t (&table)[256], uint32_t initial, const void *buf, size_t len) {
+        uint32_t c = initial ^ 0xFFFFFFFF;
+        const uint8_t *u = static_cast<const uint8_t *>(buf);
+        for (size_t i = 0; i < len; ++i) {
+            c = table[(c ^ u[i]) & 0xFF] ^ (c >> 8);
+        }
+        return c ^ 0xFFFFFFFF;
+    }
+};
```

Comment:
you could also just use `_mm_crc32_u8` I guess.
In any case, we should add some tests for the CRC calc, and also define exactly which variant we are using.
Not sure which on of these this one is https://reveng.sourceforge.io/crc-catalogue/17plus.htm#crc.cat-bits.32

Reply:
Lets improve this in another PR then?